### PR TITLE
Delete Old Version, Promote Version

### PR DESCRIPTION
**Pull request by marek-vysoky**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

**Comment by cburnette** [comment]
@coolboy A lot of changes plus some new functionality so need to review.

File: Box.V2/Managers/BoxFilesManager.cs [hunk 1]
```diff
@@ -646,5 +646,52 @@ private int GetTimeDelay(HttpResponseHeaders headers)
...
             return response.ResponseObject;
         }
+
+        /// <summary>
+        /// Discards a specific file version to the trash.
+        /// </summary>
+        /// <param name="id">Id of the file (Required).</param>
+        /// <param name="versionId">Id of the version (Required).</param>
+        /// <param name="etag">The etag of the file. This is in the ‘etag’ field of the file object.</param>
+        /// <returns>True, if version is deleted..</returns>
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: duplicate .. at the end

File: Box.V2/Managers/BoxFilesManager.cs [hunk 2]
```diff
@@ -646,5 +646,52 @@ private int GetTimeDelay(HttpResponseHeaders headers)
...
+
+            IBoxResponse<BoxFile> response = await ToResponseAsync<BoxFile>(request).ConfigureAwait(false);
+
+            return response.Status == ResponseStatus.Success;
+        }
+
+        /// <summary>
+        /// If there are previous versions of this file, this method can be used to promote one of the older versions to the top of the stack. 
+        /// This actually mints a copy of the old version and puts it on the top of the versions stack. 
+        /// The file will have the exact same contents, the same SHA1/etag, and the same name as the original.
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: The file will have the exact same contents, the same SHA1/etag, and the same name as the original. 

I am thinking "as the promoted version"

@cburnette do we need to update our docs.box.com also?

**Comment by marek-vysoky** [comment]
@coolboy @cburnette I've updated comments and resolved conflicts

**Comment by coolboy** [comment]
looks good to me, @cburnette do you think it's ok to merge?

**Pull request merged**